students/slawinski_amadeusz/lesson_03_functions/calculator.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+
+
+def print_usage():
+    """Prints how to use a calculator."""
+    print("Welcome to badly organized calculator:")
+    print("a - add")
+    print("s - subtract")
+    print("m - multiply")
+    print("d - divide")
+    print("p - power")
+    print("h,? - help")
+    print("q - QUIT")
+
+
+def op_add(var_1, var_2):
+    """Adds two numbers
+ arguments:
+    var_1 -- first number
+    var_2 -- second number
+    """
+    return var_1 + var_2
+
+
+def op_sub(var_1, var_2):
+    """Substracts two numbers
+
+    Keyword arguments:
+    var_1 -- first number
+    var_2 -- second number
+    """
+    return var_1 - var_2
+
+
+def op_mul(var_1, var_2):
+    """Multiplicates two numbers
+
+    Keyword arguments:
+    var_1 -- first number
+    var_2 -- second number
+    """
+    return var_1 * var_2
+
+
+def op_div(var_1, var_2):
+    """Divides two numbers
+
+    Keyword arguments:
+    var_1 -- divided number
+    var_2 -- dividing number
+    """
+    return var_1 / var_2
+
+
+def op_pow(var_1, var_2):
+    """Raises a number to power
+
+    Keyword arguments:
+    var_1 -- raised number
+    var_2 -- power number
+    """
+    return var_1 ** var_2
+
+
+def get_numbers():
+    """Gets two number from user."""
+    while True:
+        try:
+            print("Input 1st operand:")
+            var_1 = int(input())
+            print("Input 2nd operand:")
+            var_2 = int(input())
+            break
+        except ValueError:
+            print("Try with a number instead!")
+    return (var_1, var_2)
+
+
+def do_operation(operation, var_1, var_2):
+    print("Result:")
+    print(operation(var_1, var_2))
+
+
+print_usage()
```

Comment:
całą tą część kodu, która już wywołuje kalkulator można by zamknąć w warunku:
`if name == 'main':`
Dzięki czemu podczas importu funkcji z tego modułu do innego modułu/skryptu ta część modułu nie zostanie wykonana.
On się wykona tylko wtedy, gdy ten moduł/skrypt zostanie uruchomiony jak skrypt (na najwyższym poziomie).
Jeżeli uruchomisz ten skrypt zmienna name przyjmie wartość 'main', jeżeli zaimportujesz jako moduł to name ma wartość: 'nazwa_pliku'.
Pamiętaj o tym tworząc nowy moduł i możesz to tez dodać do pozostałych zadań.
W skryptach z lekcji 0 uzyty był ten mechanizm, możesz zobaczyć jak go użyto.